src/flashblocks.rs

```diff
@@ -0,0 +1,304 @@
+                            metadata.receipts.values().cloned().collect()
+                        } else {
+                            let existing = cache_clone
+                                .get::<Vec<OpReceipt>>(&format!("diff:receipts:{:?}", block_number))
+                                .expect("failed to get pending receipts from cache");
+                            metadata
+                                .receipts
+                                .values()
+                                .cloned()
+                                .chain(existing)
+                                .collect()
+                        };
+                        cache_clone
+                            .set(
+                                &format!("diff:receipts:{:?}", block_number),
+                                &receipts,
+                                Some(10),
+                            )
+                            .expect("failed to set pending receipts in cache");
+
+                        let execution_payload: ExecutionPayloadV3 = ExecutionPayloadV3 {
+                            blob_gas_used: 0,
+                            excess_blob_gas: 0,
+                            payload_inner: ExecutionPayloadV2 {
+                                withdrawals,
+                                payload_inner: ExecutionPayloadV1 {
+                                    parent_hash: base.parent_hash,
+                                    fee_recipient: base.fee_recipient,
+                                    state_root: diff.state_root,
+                                    receipts_root: diff.receipts_root,
+                                    logs_bloom: diff.logs_bloom,
+                                    prev_randao: base.prev_randao,
+                                    block_number: base.block_number,
+                                    gas_limit: base.gas_limit,
+                                    gas_used: diff.gas_used,
+                                    timestamp: base.timestamp,
+                                    extra_data: base.extra_data,
+                                    base_fee_per_gas: base.base_fee_per_gas,
+                                    block_hash: diff.block_hash,
+                                    transactions,
+                                },
+                            },
+                        };
+
+                        let block: OpBlock = execution_payload
+                            .try_into_block()
+                            .expect("failed to convert execution payload to block");
+
+                        // "pending" because users query the block using "pending" tag
+                        // This is an optimistic update will likely need to tweak in the future
+                        cache_clone
+                            .set("pending", &block, Some(10))
+                            .expect("failed to set block in cache");
```

Comment:
I feel like panicking in all these cases may not be ideal, as it'll crash the program I believe.